desktop/src/store/Shift/selectors.jsx

```diff
@@ -22,9 +22,15 @@ export const getCurrentShift = createSelector(
   getShiftsFromEntities,
   getShiftsFromResults,
   getShiftFromState,
-  (shifts, results, shift) => {
+  getEmployeeFromState,
+  (shifts, results, employee) => {
+
     if (!results || results.length === 0) return null
-    return shifts[shift.current.id]
+
+    let employeesHalfShifts = Object.values(shifts).filter(s => s.employeeId === employee.current.id && s.clockOutDate === null);
+    let mostRecentHalfShift = employeesHalfShifts.sort((a,b) => { return ( new Date (b.clockInDate) - new Date (a.clockInDate) ) })[0];
```

Comment:
Nice work!

Reply:
It happened to be a partial fix, found rest of the problem, fixed in this PR.